sunpy/net/dataretriever/sources/rhessi.py

```diff
@@ -1,16 +1,133 @@
 # Author: Rishabh Sharma <rishabh.sharma.gunner@gmail.com>
 # This module was developed under funding provided by
 # Google Summer of Code 2014
+import socket
+import posixpath
+from datetime import datetime
+from urllib.error import URLError
+from urllib.request import urlopen, urlretrieve
 
+from dateutil.rrule import rrule, MONTHLY
+
+import astropy.units as u
+
+from sunpy.time import TimeRange, parse_time
 from sunpy.instr import rhessi
 
 from ..client import GenericClient
 
 __all__ = ['RHESSIClient']
 
+data_servers = ('https://hesperia.gsfc.nasa.gov/hessidata/',
+                'http://hessi.ssl.berkeley.edu/hessidata/',
+                'http://soleil.i4ds.ch/hessidata/')
+
+lc_linecolors = ('black', 'pink', 'green', 'blue', 'brown', 'red',
+                 'navy', 'orange', 'green')
+
+
+def get_base_url():
+    """
+    Find the first mirror which is online
+    """
+    for server in data_servers:
+        try:
+            urlopen(server, timeout=1)
+            return server
+        except (URLError, socket.timeout):
+            pass
+
+    raise IOError('Unable to find an online HESSI server from {0}'.format(data_servers))
+
 
 class RHESSIClient(GenericClient):
 
+    def get_observing_summary_filename(self, time_range):
```

Comment:
I thought we had this done using the scraper..

Reply:
We had? I don't remember that. You got a link?

Reply:
I think it's on one of the automated closed PRs... I got a list to work on these... so one day (hopefully before 1.0).